### DISCUSSION TITLE
UI

### DISCUSSION THREAD

**Discussion by gfaust82** [pachterlab/colosseum#6] [Q&A]
Hello,

I am attempting to use your code to run the equipment as described in your repository, but the UI stalls at the PYQT5 popup "Connecting to Arduino at port:..."

I am running the script in Python 3.10 on Ubuntu 22.04.

Any help would be much appreciated.

**Comment by annekil0530** [comment]
Hi gfaust82,

Thanks for bringing this up! I am so sorry you're having this issue. Would you mind trying the web-based UI at https://pachterlab.github.io/colosseum/ (link also included in the README of the repo)? This UI is more up-to-date and does not require the installation of the local UI python files. You would just upload the .ino code onto the Arduino and start up the UI by going to the link above, and the same UI as the python UI should start up. Please let us know if it works!

Best,
Anne

**Reply by gfaust82** [reply]
Hello Anne,

Thanks for replying to my issue. I am not able to connect to the device using the web-based UI in Chrome for Ubuntu. I receive either a ‘Connection timed out’ or a ‘Serial Port is already open’ error message. 

I’m Chrome on Windows 10, I am able to connect once, but then I have to restart the PC to connect a second time to the device. 

Any advice is much appreciated. 

Best,
Marty

**Reply by annekil0530** [reply]
Ah I may have an idea of what's going on... After you upload the Arduino sketch to the Arduino, could you shut down the Arduino IDE program and then run the UI? Please let me know if this does anything!

Best,
Anne

**Reply by gfaust82** [reply]
Hi Anne,

I didn’t have Arduino IDE open, but I still wasn’t able to connect to the Arduino. The good news is I was able to get the local Python package to work by moving run.py one directory level up (to the SOFTWARE directory). We are thinking of creating a version of the software with timing based on the “inlet start” signal of an HPLC system. Please let me know if you would like to work together on this. 

Best,
Marty

**Reply by annekil0530** [reply]
Hi Marty,

I apologize for the late follow-up on this!

Please email me at <ykil@caltech.edu> to discuss things further--I would love to help!

Best,
Anne
